src/model/test/CoilHeatingElectricMultiStage_GTest.cpp

```diff
@@ -0,0 +1,70 @@
+*  disclaimer in the documentation and/or other materials provided with the distribution.
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include "ModelFixture.hpp"
+
+#include "../CoilHeatingElectricMultiStage.hpp"
+#include "../CoilHeatingElectricMultiStage_Impl.hpp"
+#include "../CoilHeatingElectricMultiStageStageData.hpp"
+#include "../CoilHeatingElectricMultiStageStageData_Impl.hpp"
+#include "../ScheduleConstant.hpp"
+#include "../ScheduleConstant_Impl.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, CoilHeatingElectricMultiStage_GettersSetters) {
+  Model m;
+  CoilHeatingElectricMultiStage coil(m);
+
+  EXPECT_FALSE(coil.availabilitySchedule());
+  EXPECT_EQ(0u, coil.stages().size());
+
+  ScheduleConstant scheduleConstant(m);
+  EXPECT_TRUE(coil.setAvailabilitySchedule(scheduleConstant));
+
+  ASSERT_TRUE(coil.availabilitySchedule());
+  EXPECT_EQ(scheduleConstant.handle(), coil.availabilitySchedule().get().handle());
+
+  coil.resetAvailabilitySchedule();
+
+  EXPECT_FALSE(coil.availabilitySchedule());
+}
+
+TEST_F(ModelFixture, CoilHeatingElectricMultiStage_Remove) {
+  Model m;
+  CoilHeatingElectricMultiStage coil(m);
+  CoilHeatingElectricMultiStageStageData stage(m);
+  coil.addStage(stage);
+  coil.remove();
+
+  EXPECT_EQ(0u, m.modelObjects().size());
+}
```

Comment:
I want to see:
* A test for adding it on an AirLoopHVAC directly (do we allow or don't we?!)
* Definitely a clone test (in same AND other model) with respect to the StageData

Reply:
What gives you the impression it can be added on an AirLoopHVAC directly? (AFAIK it can be placed on AirLoopHVAC:UnitarySystem.)

Reply:
https://bigladdersoftware.com/epx/docs/22-1/input-output-reference/group-heating-and-cooling-coils.html#inputs-4-015

>If the coil is used in the air loop simulation directly on a branch and is temperature controlled using a Set Point Manager, then the node that is the control node needs to be specified here. If the coil is used in an air terminal unit or other parent object (ZoneHVAC: or AirloopHVAC:), the coil is controlled by the parent object and the temperature set point node name is not required. At present, the multistage electric heating coil does not model temperature setpoint control.

Reply:
Apparently BOTH the IDD and the I/O are wrong per some quick test I've done modifying an example file to try to put a Coil:heating:Electric:MultiStage on an AirLoopHVAC branch